### DISCUSSION TITLE
Preview functions are called in the non preview functions

### DISCUSSION THREAD

**Discussion by mbalajee** [cqfn/diKTat#1725] [Ideas]
It will be really useful to have a rule to check if Preview functions are called in the non preview functions

**Comment by orchestr7** [comment]
@mbalajee what do you mean by `Preview functions`? Can you please send a link to Kotlin spec?

**Comment by orchestr7** [comment]
You mean this one: https://github.com/saveourtool/diktat/issues/1322 ?

**Reply by mbalajee** [reply]
I mean the composable preview functions.  [#1322](https://github.com/saveourtool/diktat/issues/1322) checks for suffix and modifier.  Google [suggests](https://developer.android.com/jetpack/compose/tooling/previews) to create a separate composable for previews.  Since previews contain hardcoded values it would be helpful to have a rule that ensures that these preview composables are not called in a non-preview composable
